service/sharddistributor/executorclient/syncgeneric/map.go

```diff
@@ -0,0 +1,48 @@
+package syncgeneric
+
+import "sync"
+
+// Map is a wrapper around sync.Map that provides a more convenient, generic,
+// and type-safe API.
+type Map[K comparable, V any] struct {
+	contents sync.Map
```

Comment:
nit: maybe a map and a mutext can do better :) but we can experiment with this later